scripts/install/install_unix.sh

```diff
@@ -13,165 +13,638 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# shellcheck disable=SC2119
+# SC2119 -> Use foo "$@" if function's $1 should mean script's $1.
```

Comment:
I dislike having this global disable; Is this violated all over, or can we disable it for just where this is relevant?

Reply:
I am not sure, my suspicion is that it has always been there, looking at some of our past agents / collectors. 

I am generally a fan of disabling shellcheck rules where they happen instead of globally.

I just ran shellcheck against the script with those two lines removed and it passes fine (it has two unrelated issues)
```
In scripts/install/install_unix.sh line 260:
      while [ -z "$proxy_password" ] && [ ! "$accept_defaults" = "yes" ]; do
                                             ^--------------^ SC2154: accept_defaults is referenced but not assigned.


In scripts/install/install_unix.sh line 297:
      os="linux"
      ^-- SC2034: os appears unused. Verify use (or export if used externally).
```

Reply:
It might be a good idea to have shellcheck run as part of our workflow. The check is super fast and won't have a big impact on CI.

Reply:
I think if we have more shell scripts I would do so but we have only a few that likely won't change much I don't think it's worth the extra CI and tooling right now.

Reply:
It sounds like we're OK with these two lines then. We don't check against it in CI, but I know a lot of us use the shell-check extension in vscode to catch mistakes.

Reply:
Yeah I think that's fine for now.